daemons/compute/server/servers/server_default.go

```diff
@@ -308,6 +308,30 @@ Retry:
 	}, nil
 }
 
+func (s *defaultServer) List(ctx context.Context, req *pb.DataMovementListRequest) (*pb.DataMovementListResponse, error) {
+
+	// Only get the DMs that match the workflow and workflow namespace
+	opts := []client.ListOption{
+		client.MatchingLabels{"dws.cray.hpe.com/owner.namespace": req.Namespace},
+		client.MatchingLabels{"dws.cray.hpe.com/owner.name": req.Workflow},
+	}
+
+	list := dmv1alpha1.RsyncNodeDataMovementList{}
+	if err := s.client.List(ctx, &list, opts...); err == nil {
+		rsyncdm_uids := []string{}
```

Comment:
Consider allocating the array upfront with `make([]string, len(list.Items))` and void the append (see comment in `server_sim.go`). Small optimization.